sdk/python/kfp_tekton/compiler/compiler.py

```diff
@@ -170,13 +182,111 @@ def _group_to_dag_template(self, group, inputs, outputs, dependencies, pipeline_
 
       operand1_value = self._resolve_value_or_reference(condition.operand1, subgroup_inputs)
       operand2_value = self._resolve_value_or_reference(condition.operand2, subgroup_inputs)
-
       template['kind'] = 'Condition'
       template['spec']['params'] = [
         {'name': 'operand1', 'value': operand1_value},
         {'name': 'operand2', 'value': operand2_value},
         {'name': 'operator', 'value': str(condition.operator)}
       ]
+
+    # dsl does not expose Graph so here use sub_group.type to check whether it's graph
+    if sub_group.type == "graph":
+      # for graph now we just support as a pipeline loop with just 1 iteration
+      loop_args_name = "just_one_iteration"
+      loop_args_value = ["1"]
+
+      # Special handling for recursive subgroup
+      if sub_group.recursive_ref:
+        # generate ref graph name
+        tmp_group_names = [pipeline_name, group_type, sanitize_k8s_name(sub_group.name)]
+        if self.uuid:
+          tmp_group_names.insert(1, self.uuid)
+        ref_group_name = '-'.join(tmp_group_names)
+
+        # generate params
+        params = [{
+          "name": loop_args_name,
+          "value": loop_args_value
+        }]
+
+        # get other input params, for recursion need rename the param name to the refrenced one
+        for i in range(len(sub_group.inputs)):
+            input = sub_group.inputs[i]
+            inputRef = sub_group.recursive_ref.inputs[i]
+            if input.op_name:
+              params.append({
+                'name': inputRef.full_name,
+                'value': '$(tasks.%s.results.%s)' % (input.op_name, input.name)
+              })
+            else:
+              params.append({
+                'name': inputRef.full_name, 'value': '$(params.%s)' % input.name
+              })
+            
+        self.recursive_tasks.append({
+          'name': sub_group.name,
+          'taskRef': {
+            'apiVersion': 'tekton.dev/v1beta1',
+            'kind': 'Pipeline',
```

Comment:
oh for recursion over here we should use pipelineloop

Reply:
Good catch!